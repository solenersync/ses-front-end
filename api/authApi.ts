export const createUser = async (userData: any) => {
  let user = {
    email: userData.email,
    name: userData.name,
    password: userData.password,
  };
  const res = await fetch("http://localhost:8081/api/v1/users/user/create", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(user),
  });
  return res;
};

export const authenticate = async (userData: any) => {
  let userDetails = {
    email: userData.email,
    password: userData.password,
  };
  const res = await fetch("http://localhost:8081/api/v1/users/user/authenticate", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(userDetails),
  });
  return res.json();
};

