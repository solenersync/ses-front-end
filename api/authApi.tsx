export const createUser = async (userData: any) => {
  let user = {
    email: userData.email,
    name: userData.name,
    password: userData.password,
  };
  const res = await fetch("http://localhost:8080/api/v1/users/user/create", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(user),
  });
  return res;
};

export const authenticate = async (userData: any) => {
  let user = {
    email: userData.email,
    password: userData.password,
  };
  const res = await fetch("http://localhost:8080/api/v1/users/user/authenticate", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(user),
  });
  return res.json();
};

