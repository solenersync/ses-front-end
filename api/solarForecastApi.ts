import { SolarArray } from 'models/SolarArray';

export const getSolarForecast = async (arrayData: SolarArray) => {

  const res = await fetch("http://localhost:8080/api/v1/pv/daily", {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(arrayData),
  });
  return res.json();
};
