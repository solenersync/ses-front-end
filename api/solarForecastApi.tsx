export const getSolarForecast =  async () => {

 let userData = {
  "user_id": 1,
  "lat": 52.12121,
  "lon": 8.121,
  "peakpower": 6,
  "loss": 0.1,
  "angle": 35,
  "aspect": 0,
  "mounting": "FREE",
  "month": 4
}

const res = await fetch(
  'http://localhost:8080/api/v1/pv/daily',
  {
    method: 'POST',
    headers: {},
    body: JSON.stringify(userData),
  }
);
return res;
}