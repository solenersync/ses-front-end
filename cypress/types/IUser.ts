export class IUser {
 email: string;
 password: string;
 name: string;
}
