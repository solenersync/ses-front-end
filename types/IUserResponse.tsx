export interface IUserResponse {
  email: string;
  registeredDate: string;
  name: string;
  userId: number;
}
