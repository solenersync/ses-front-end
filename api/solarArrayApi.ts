import { ISolarArray } from 'types/ISolarArray';
import { axiosSolarArrayApi } from 'axios.config';


export const getArrayData = async (userId: string) => {
  axiosSolarArrayApi.defaults.baseURL = process.env.API_BASE_URL ?? axiosSolarArrayApi.defaults.baseURL;
  try {
    const response = await axiosSolarArrayApi.get(`/api/v1/solar-arrays/array/user/${userId}`);
    return response.data;
  } catch (error) {
    return null;
  }
};

export const createArray = async (arrayData: ISolarArray) => {
  axiosSolarArrayApi.defaults.baseURL = process.env.API_BASE_URL ?? axiosSolarArrayApi.defaults.baseURL;
  const response = await axiosSolarArrayApi.post('/api/v1/solar-arrays/create', arrayData);
  return response;
};


export const updateArray = async (arrayData: ISolarArray) => {
  axiosSolarArrayApi.defaults.baseURL = process.env.API_BASE_URL ?? axiosSolarArrayApi.defaults.baseURL;
  try {
    const response = await axiosSolarArrayApi.post('/api/v1/solar-arrays/update', arrayData);
    return response;
  } catch (error) {
    return null;
  }
};

export const solarArrayApi =  {
  getArrayData,
  createArray,
  updateArray,
}