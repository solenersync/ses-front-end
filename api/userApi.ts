import { axiosUserApi } from 'axios.config';
import { ICreateUser } from '../types/ICreateUser';
import { IUpdateUser } from '../types/IUpdateUser';
import { IBasicAuthUser } from '../types/IBasicAuthUser';

export const getUser = async (email: string) => {
  axiosUserApi.defaults.baseURL = process.env.API_BASE_URL ?? axiosUserApi.defaults.baseURL;
  try {
    const response = await axiosUserApi.post('/api/v1/users/user', { email });
    return response;
  } catch (error) {
    return null;
  }
};

export const createUser = async (userData: ICreateUser) => {
  axiosUserApi.defaults.baseURL = process.env.API_BASE_URL ?? axiosUserApi.defaults.baseURL;
  try {
    const response = await axiosUserApi.post('/api/v1/users/user/create', userData);
    console.log(response.status);
    return response;
  } catch (error) {   
    if (error.response && error.response.status === 409) {
      throw error;
    }
    return null;
  }
};

export const updateUser = async (userData: IUpdateUser) => {
  axiosUserApi.defaults.baseURL = process.env.API_BASE_URL ?? axiosUserApi.defaults.baseURL;
  try {
    const response = await axiosUserApi.put('/api/v1/users/user/update', userData);
    return response;
  } catch (error) {
    return null;
  }
};

export const authenticate = async (userData: IBasicAuthUser) => {
  axiosUserApi.defaults.baseURL = process.env.API_BASE_URL ?? axiosUserApi.defaults.baseURL;
  try {
    const response = await axiosUserApi.post('/api/v1/users/user/authenticate', userData);
    return response;
  } catch (error) {
    return null;
  }
};


