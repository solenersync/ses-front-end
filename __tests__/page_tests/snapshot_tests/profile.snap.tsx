import { render, act } from '@testing-library/react';
import Profile from 'pages/profile';
import { useSession as originalUseSession } from 'next-auth/react';
import { User } from 'next-auth';
import { getUser } from 'api/userApi';
import '@testing-library/jest-dom';
import { IUserResponse } from 'types/IUserResponse';
import { AxiosResponse } from 'axios';

jest.mock('api/userApi');
jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));
jest.mock('next-auth/react', () => ({
  ...jest.requireActual('next-auth/react'),
  useSession: jest.fn()
}));

describe('Profile page', () => {
  const user: User = { name: 'John Doe', email: 'jd@test.com',  userId: '1', id:'' };
  const userRespBody: IUserResponse = { name: 'John Doe', email: 'jd@test.com',  userId: 1, registeredDate:'2023-03-16T10:40:30' }

  const axiosResponse: AxiosResponse = {
    data: userRespBody,
    status: 200,
    statusText: '',
    headers: undefined,
    config: undefined
  };

  beforeEach(() => {
    (getUser as jest.Mock).mockResolvedValue(axiosResponse);
    (originalUseSession as jest.Mock).mockReturnValue({ status: 'authenticated', data: {user}});
  });

  test('renders the profile page', async () => {
    let container: HTMLElement;

    await act(async () => { 
      const result = render(<Profile />);
      container = result.container;
    });  

    expect(container).toMatchSnapshot();
  });
  
});
