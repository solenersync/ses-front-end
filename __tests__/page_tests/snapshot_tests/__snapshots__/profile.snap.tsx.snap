// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page renders the profile page 1`] = `
<div>
  <form
    action="#"
    class="mt-12 px-6 lg:px-8"
    method="POST"
  >
    <div
      class="space-y-8 divide-y divide-gray-200 sm:space-y-5"
    >
      <div
        class="space-y-6 pt-8 sm:space-y-5 sm:pt-10"
      >
        <div>
          <h3
            class="text-lg font-medium leading-6 text-gray-900"
          >
            Personal Information
          </h3>
        </div>
        <div
          class="space-y-6 sm:space-y-5"
        >
          <div
            class="sm:grid sm:grid-cols-3 sm:items-start sm:gap-4 sm:border-t sm:border-gray-200 sm:pt-5"
          >
            <label
              class="block text-sm font-medium text-gray-700 sm:mt-px sm:pt-2"
              for="name"
            >
              Name
            </label>
            <div
              class="mt-1 sm:col-span-2 sm:mt-0"
            >
              <input
                autocomplete="given-name"
                class="block w-full max-w-lg rounded-md border-gray-300 shadow-sm focus:border-rose-500 focus:ring-rose-500 sm:max-w-xs sm:text-sm"
                id="name"
                name="name"
                type="text"
                value="John Doe"
              />
            </div>
          </div>
          <div
            class="sm:grid sm:grid-cols-3 sm:items-start sm:gap-4 sm:border-t sm:border-gray-200 sm:pt-5"
          >
            <label
              class="block text-sm font-medium text-gray-700 sm:mt-px sm:pt-2"
              for="email"
            >
              Email address
            </label>
            <div
              class="mt-1 sm:col-span-2 sm:mt-0"
            >
              <input
                autocomplete="email"
                class="block w-full appearance-none rounded-md border border-gray-300 px-3 py-2 placeholder-gray-400 shadow-sm cursor-not-allowed focus:outline-none sm:text-sm"
                disabled=""
                id="email"
                name="email"
                type="text"
                value="jd@test.com"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pt-5"
    >
      <div
        class="flex justify-end"
      >
        <a
          href="/dashboard"
        >
          <button
            class="rounded-md border border-gray-300 bg-white py-2 px-4 text-sm font-medium text-gray-700 shadow-sm hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-rose-500 focus:ring-offset-2"
            type="button"
          >
            Cancel
          </button>
        </a>
        <button
          class="ml-3 inline-flex justify-center rounded-md border border-transparent bg-rose-600 py-2 px-4 text-sm font-medium text-white shadow-sm hover:bg-rose-700 focus:outline-none focus:ring-2 focus:ring-rose-500 focus:ring-offset-2"
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;
